File: src/lib/sports.ts
// Single source of truth for the sports the platform lists. Only Cricket is
// wired up today — every other sport renders as "Coming soon" until it has
// its own flow. Give a sport a `to` to turn its card live.

export interface Sport {
  id: string
  name: string
  icon: string
  to?: string
}

export const SPORTS: Sport[] = [
  { id: 'cricket', name: 'Cricket', icon: '🏏', to: '/home' },
  { id: 'football', name: 'Football', icon: '⚽' },
  { id: 'rugby', name: 'Rugby', icon: '🏉' },
  { id: 'hockey', name: 'Hockey', icon: '🏑' },
  { id: 'tennis', name: 'Tennis', icon: '🎾' },
  { id: 'basketball', name: 'Basketball', icon: '🏀' },
  { id: 'badminton', name: 'Badminton', icon: '🏸' },
  { id: 'volleyball', name: 'Volleyball', icon: '🏐' },
  { id: 'baseball', name: 'Baseball', icon: '⚾' },
  { id: 'kabaddi', name: 'Kabaddi', icon: '🤼' },
  { id: 'golf', name: 'Golf', icon: '⛳' },
  { id: 'tableTennis', name: 'Table Tennis', icon: '🏓' },
  { id: 'bikeRacing', name: 'Bike Racing', icon: '🏍️' },
  { id: 'carRacing', name: 'Car Racing', icon: '🏎️' },
  { id: 'f1Racing', name: 'F1 Racing', icon: '🏁' },
]

// Every auction created before the `sport` field existed, and every one
// created without an explicit choice, is a Cricket auction.
export const DEFAULT_SPORT_ID = 'cricket'

export function sportName(id: string | null | undefined): string {
  return SPORTS.find((s) => s.id === id)?.name ?? 'Cricket'
}

File: android/app/src/main/java/com/mplauction/android/data/Sports.kt
package com.mplauction.android.data

// Mirrors src/lib/sports.ts — single source of truth for the sports the
// platform lists. Only Cricket is wired up; the rest render "Coming soon"
// until they have their own flow, matching the web app.
data class Sport(val id: String, val name: String, val icon: String, val wired: Boolean = false)

val SPORTS =
  listOf(
    Sport("cricket", "Cricket", "🏏", wired = true),
    Sport("football", "Football", "⚽"),
    Sport("rugby", "Rugby", "🏉"),
    Sport("hockey", "Hockey", "🏑"),
    Sport("tennis", "Tennis", "🎾"),
    Sport("basketball", "Basketball", "🏀"),
    Sport("badminton", "Badminton", "🏸"),
    Sport("volleyball", "Volleyball", "🏐"),
    Sport("baseball", "Baseball", "⚾"),
    Sport("kabaddi", "Kabaddi", "🤼"),
    Sport("golf", "Golf", "⛳"),
    Sport("tableTennis", "Table Tennis", "🏓"),
    Sport("bikeRacing", "Bike Racing", "🏍️"),
    Sport("carRacing", "Car Racing", "🏎️"),
    Sport("f1Racing", "F1 Racing", "🏁"),
  )

const val DEFAULT_SPORT_ID = "cricket"

fun sportName(id: String?): String = SPORTS.find { it.id == id }?.name ?: "Cricket"
